app-service-environment-serviceplan-logicapp/azuredeploy.json

```diff
@@ -8,6 +8,12 @@
         "description": "The name to use for the new AppService Environment"
       }
     },
+    "location": {
```

Comment:
If i recall correctly the network should be already in place and is used as a resource reference. (What can be outside the RG you deploy this template to). The subnet should be a dedicated worker subnet only no other vm's should live there.